### ISSUE TITLE
Stop importing deprecated github.com/golang/protobuf

### ISSUE THREAD

**Issue by bwplotka** [prometheus/client_golang#1175]
AC:
* Move to https://pkg.go.dev/google.golang.org/protobuf API

See https://github.com/prometheus/common/issues/317

**Comment by bwplotka** [comment]
Fixed by https://github.com/prometheus/client_golang/pull/1183

**Comment by zhsj** [comment]
@bwplotka since github.com/golang/protobuf is still in go.mod, thus I didn't close it by commit message. Probably keep it open until the tests have migrated as well?

**Comment by SuperQ** [comment]
I agree, we still need to clean up the tests so the Go modules dependencies are fixed.

**Comment by drobazko** [comment]
Hey folks.
Any updates on this issue?

**Comment by rufreakde** [comment]
@SuperQ any updates?

**Comment by ash2k** [comment]
I found https://github.com/prometheus/common/pull/509 that should help remove the deprecated proto package from the transitive dependency list.

**Comment by ash2k** [comment]
Another chain of dependencies is via the `github.com/prometheus/common` too:

```
go mod why -m github.com/golang/protobuf
# github.com/golang/protobuf
github.com/prometheus/common/config
golang.org/x/oauth2
golang.org/x/oauth2/internal
google.golang.org/appengine/urlfetch
github.com/golang/protobuf/proto
```

Work to address that is tracked in https://github.com/golang/appengine/issues/228. After it's done, someone will need to update golang.org/x/oauth2 to use the new version (there is a `google.golang.org/appengine/v2` already, so a major version bump will be needed).
